crates/ntx-builder/src/transaction.rs

```diff
@@ -75,58 +75,44 @@ pub struct NtxContext {
+    pub fn execute_transaction(
+        self,
+        tx: TransactionCandidate,
+    ) -> impl FutureMaybeSend<NtxResult<()>> {
         let TransactionCandidate {
             account,
             notes,
             chain_tip_header,
             chain_mmr,
         } = tx;
-
         tracing::Span::current().set_attribute("account.id", account.id());
         tracing::Span::current().set_attribute("notes.count", notes.len());
         tracing::Span::current()
             .set_attribute("reference_block.number", chain_tip_header.block_num());
 
-        // Work-around for `TransactionExecutor` not being `Send`.
-        tokio::task::spawn_blocking(move || {
-            {
-                {
-                    let rt = tokio::runtime::Builder::new_current_thread()
-                        .enable_all()
-                        .build()
-                        .expect("runtime should be built");
-
-                    rt.block_on(
-                        async move {
-                            let mut notes = notes
-                                .into_iter()
-                                .map(|note| InputNote::Unauthenticated { note: note.into() })
-                                .collect::<Vec<_>>();
-                            // We shuffle the notes here to prevent having a failing note always in
-                            // front.
-                            notes.shuffle(&mut rand::rng());
-                            let notes = InputNotes::new(notes).map_err(NtxError::InputNotes)?;
+        async move {
+            async move {
+                let mut notes = notes
+                    .into_iter()
+                    .map(|note| InputNote::Unauthenticated { note: note.into() })
+                    .collect::<Vec<_>>();
+                // We shuffle the notes here to prevent having a failing note always in
+                // front.
+                notes.shuffle(&mut rand::rng());
+                let notes = InputNotes::new(notes).map_err(NtxError::InputNotes)?;
 
-                            let data_store =
-                                NtxDataStore::new(account, chain_tip_header, chain_mmr);
+                let data_store = NtxDataStore::new(account, chain_tip_header, chain_mmr);
 
-                            let notes = Box::pin(self.filter_notes(&data_store, notes)).await?;
-                            let executed = Box::pin(self.execute(&data_store, notes)).await?;
-                            let proven = Box::pin(self.prove(executed)).await?;
-                            self.submit(proven).await?;
-                            Ok(())
-                        }
-                        .in_current_span(),
-                    )
-                }
+                let notes = Box::pin(self.filter_notes(&data_store, notes)).await?;
+                let executed = Box::pin(self.execute(&data_store, notes)).await?;
+                let proven = Box::pin(self.prove(executed)).await?;
```

Comment:
Question: is there an alternative to these `Box::pin()` or do we have to keep using them from now on?

Reply:
We do get clippy warnings due to future sizes, ``Box::pin` shifts pieces of it to the heap, which makes the wrapping future smaller and quiets the clippy warnings. Out future sizes exceeded 100 kBytes, which is already pretty heavy, mostly from having to carry plenty of stack residing state across await points.